internal/config/tls_credentials.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+// TLSCredentials holds the fields for TLS credentials
+// that are used for starting a server.
+type TLSCredentials struct {
+	// CertFile is the file path containing the TLS certificate.
+	CertFile string `yaml:"cert_file"`
+
+	// KeyFile is the file path containing the TLS key.
+	KeyFile string `yaml:"key_file"`
+}
+
+func (tc *TLSCredentials) nonEmpty() bool {
+	return tc != nil && tc.CertFile != "" && tc.KeyFile != ""
```

Comment:
Perhaps having an OR between the cert files is better: if there is a typo then it should end up being reported as an error when building the credentials, which hopefully indicates that one of the files is empty.

Reply:
Sorry I don't follow. Are you asking that we make it `tc != nil && (tc.CertFile != "" || tc.KeyFile != "")`?
If so, gotcha, yeah that makes sense and it will produce an error.